### PR TITLE
Don’t require unique email addresses for forms with captcha.

### PR DESCRIPTION
**Pull request by torotil**

### CODE REVIEW

File: campaignion_wizard/src/WebformStepUnique.php [hunk 1]
```diff
@@ -30,4 +30,26 @@ public function validateStep($form, &$form_state) {
...
+   *   The node which’s captcha settings should be checked.
+   *
+   * @return bool
+   *   TRUE if a captcha is set up for this node’s webform otherwise FALSE.
+   */
+  protected function hasCaptcha($node) {
+    if (!module_exists('captcha')) {
+      return FALSE;
+    }
+    $form_id = "webform_client_form_{$node->nid}";
```
Comment (katzmo) [inline campaignion_wizard/src/WebformStepUnique.php:47]: Is that save? What if there are several forms on the same nid?

File: campaignion_wizard/src/WebformStepUnique.php [hunk 2]
```diff
@@ -20,7 +20,7 @@ public function validateStep($form, &$form_state) {
         $c = reset($cs);
       }
     }
-    if ($c) {
+    if ($c && !$this->hasCaptcha($form['#node'])) {
```
Comment (katzmo) [inline campaignion_wizard/src/WebformStepUnique.php:24]: I don’t quite get the logic behind "if there is a capture, the email address does not have to be unique", aren’t those completely different things? For spamming a form, one could easily generate unique email addresses. On the other hand a capture won’t stop real people from submitting the form twice with the same data.

Comment (torotil) [reply campaignion_wizard/src/WebformStepUnique.php:24]: True. Anyway we require only at least one of them to be set up even if that’s not a comprehensive solution.

File: campaignion_wizard/src/WebformStepUnique.php (hunk 1 above)
Comment (torotil) [reply campaignion_wizard/src/WebformStepUnique.php:47]: How can there be more than one webform for a webform node?

Comment (katzmo) [reply campaignion_wizard/src/WebformStepUnique.php:47]: Well if there can’t, it’s fine ;)

**Review by katzmo — APPROVED** [review]

**Pull request merged**